geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/WanCopyRegionCommand.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.cli.commands;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+
+import org.apache.geode.cache.execute.FunctionInvocationTargetException;
+import org.apache.geode.cache.execute.ResultCollector;
+import org.apache.geode.management.cli.CliMetaData;
+import org.apache.geode.management.cli.ConverterHint;
+import org.apache.geode.management.cli.GfshCommand;
+import org.apache.geode.management.internal.cli.functions.WanCopyRegionFunction;
+import org.apache.geode.management.internal.cli.result.model.ResultModel;
+import org.apache.geode.management.internal.functions.CliFunctionResult;
+import org.apache.geode.management.internal.i18n.CliStrings;
+import org.apache.geode.security.ResourcePermission.Operation;
+import org.apache.geode.security.ResourcePermission.Resource;
+
+public class WanCopyRegionCommand extends GfshCommand {
+  private final WanCopyRegionFunction wanCopyRegionFunction = new WanCopyRegionFunction();
+
+  @CliCommand(value = CliStrings.WAN_COPY_REGION, help = CliStrings.WAN_COPY_REGION__HELP)
+  @CliMetaData(relatedTopic = {CliStrings.TOPIC_GEODE_DATA, CliStrings.TOPIC_GEODE_REGION})
+  public ResultModel wanCopyRegion(
+      @CliOption(key = CliStrings.WAN_COPY_REGION__REGION, mandatory = true,
+          optionContext = ConverterHint.REGION_PATH,
+          help = CliStrings.WAN_COPY_REGION__REGION__HELP) String regionName,
+      @CliOption(key = CliStrings.WAN_COPY_REGION__SENDERID,
+          optionContext = ConverterHint.GATEWAY_SENDER_ID,
+          help = CliStrings.WAN_COPY_REGION__SENDERID__HELP) String senderId,
```

Comment:
This is along the same lines as Jinmei's comment below.

If I execute the command with no sender id (which is currently not mandatory), I see:
```
Executing - wan-copy region --region=/data_pr_parallel

Member | Status | Message
------ | ------ | ---------------------
ln-1   | ERROR  | Sender null not found
```
I think the sender id should be mandatory:
```
@CliOption(key = CliStrings.WAN_COPY_REGION__SENDERID, mandatory = true,
    optionContext = ConverterHint.GATEWAY_SENDER_ID,
    help = CliStrings.WAN_COPY_REGION__SENDERID__HELP) String senderId,
```
That provides a better message:
```
Executing - wan-copy region --region=/data_pr_parallel

Jul 01, 2021 1:42:01 PM org.springframework.shell.core.SimpleParser printHintMessage
WARNING: You should specify option (--sender-id, --max-rate, --batch-size, --cancel) for this command
  --sender-id=  is required
Use "help wan-copy region" (without the quotes) for detailed usage information.
```

Reply:
Now I understand. I forgot to make sender-id mandatory. Thanks to both!